sdk/include/ndk/peb_teb.h

```diff
@@ -357,11 +357,6 @@ typedef struct STRUCT(_TEB)
             UCHAR          IdealProcessor;
         };
     };
-#elif (NTDDI_VERSION >= NTDDI_LONGHORN)
```

Comment:
I don't have a Vista kernel to check with, but Win7 uses the version above in the >= NTDDI_WIN10 block. What information is your change based on?

Reply:
I not found any usage of SpareBool0-SpareBool2 in the code and from their names it looks like field stubs to keep offset for IdealProcessor field. 

Maybe comment out this NTDDI_LONGHORN block now instead of deleting?

Reply:
Okay, I checked, and this block is correct for Vista. However the WIN10 check above should say WIN7 as I mentioned.
The InDbgPrint and HasFiberData members do not exist in newer versions. The DbgInDebugPrint and DbgHasFiberData in SameTebFlags below seem to be their replacements (they don't have the Dbg- prefix though)

```
   +0x17ee SameTebFlags     : Uint2B
   +0x17ee SafeThunkCall    : Pos 0, 1 Bit
   +0x17ee InDebugPrint     : Pos 1, 1 Bit
   +0x17ee HasFiberData     : Pos 2, 1 Bit
   +0x17ee SkipThreadAttach : Pos 3, 1 Bit
   +0x17ee WerInShipAssertCode : Pos 4, 1 Bit
   +0x17ee RanProcessInit   : Pos 5, 1 Bit
   +0x17ee ClonedThread     : Pos 6, 1 Bit
   +0x17ee SuppressDebugMsg : Pos 7, 1 Bit
   +0x17ee DisableUserStackWalk : Pos 8, 1 Bit
   +0x17ee RtlExceptionAttached : Pos 9, 1 Bit
   +0x17ee InitialThread    : Pos 10, 1 Bit
   +0x17ee SpareSameTebBits : Pos 11, 5 Bits
```

This means the absence of InDbgPrint & friends is the desired behavior, as the structure must be compatible with Windows. If the kernel doesn't compile with NTDDI_VERSION=NTDDI_VISTA, I'm not sure I'd even consider that a bug.

Reply:
damn, you right( ok, roll it back

found it on http://terminus.rewolf.pl/terminus/structures/ntdll/_TEB_combined.html